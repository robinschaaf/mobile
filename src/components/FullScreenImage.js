import React from 'react'
import {
  Modal,
  TouchableOpacity,
  View
} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import ZoomableImage from './ZoomableImage'
import Icon from 'react-native-vector-icons/FontAwesome'
import StyledText from './StyledText'

class FullScreenImage extends React.Component {
  render() {
    const zoomMessage =
      <View style={styles.rowContainer}>
        <Icon name="info-circle" style={styles.infoIcon} />
        <StyledText additionalStyles={[styles.message]} text='You can zoom into this image' />
      </View>

    return (
      <Modal
        animationType={'fade'}
        transparent={true}
        onRequestClose={() => {}}
        visible={this.props.isVisible}>
        <View style={styles.container}>
          <ZoomableImage
            source={this.props.source}
            handlePress={this.props.handlePress}
            allowPanAndZoom={!!this.props.allowPanAndZoom} />
          {this.props.allowPanAndZoom ? zoomMessage : null}
          <TouchableOpacity
            activeOpacity={0.5}
            onPress={this.props.handlePress}
            style={styles.closeIcon}>
            <Icon name="times" style={styles.icon} />
          </TouchableOpacity>
        </View>
      </Modal>
    );
  }
}

const styles = EStyleSheet.create({
  container: {
    backgroundColor: 'rgba(0, 0, 0, 0.9)',
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  closeIcon: {
    backgroundColor: 'transparent',
    position: 'absolute',
    top: 13,
    right: 5
  },
  icon: {
    backgroundColor: 'transparent',
    color: 'white',
    fontSize: 24,
    padding: 15,
  },
  rowContainer: {
    backgroundColor: 'rgba(0, 0, 0, 0.9)',
    position: 'absolute',
    bottom: 10,
    left: 10,
    flexDirection: 'row',
    alignItems: 'center'
  },
  message: {
    color: 'white',
  },
  infoIcon: {
    color: '$transluscentWhite',
    fontSize: 20,
    padding: 5,
  }
})

FullScreenImage.propTypes = {
  source: React.PropTypes.object,
  isVisible: React.PropTypes.bool,
  handlePress: React.PropTypes.func,
  allowPanAndZoom: React.PropTypes.bool
}

export default FullScreenImage
