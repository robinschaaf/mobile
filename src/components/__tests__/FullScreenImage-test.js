import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import FullScreenImage from '../FullScreenImage'

const imageSource = {
  uri: 'https://placekitten.com/200/300'
}

it('renders correctly', () => {
  const tree = renderer.create(
    <FullScreenImage
      source={imageSource}
      handlePress={jest.fn}
      isVisible={true}
      allowPanAndZoom={true}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders hidden correctly', () => {
  const tree = renderer.create(
    <FullScreenImage
      source={imageSource}
      handlePress={jest.fn}
      isVisible={false}
      allowPanAndZoom={true}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
