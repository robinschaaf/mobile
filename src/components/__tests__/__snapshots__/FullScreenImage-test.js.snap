exports[`test renders correctly 1`] = `
<Modal
  animationType="fade"
  onRequestClose={[Function]}
  transparent={true}
  visible={true}>
  <View
    style={undefined}>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "height": 667,
            "width": 375,
          },
        ]
      }>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }>
        <View
          onLayout={[Function]}
          style={
            Object {
              "height": 0,
              "width": 0,
            }
          }>
          <Image
            source={
              Object {
                "uri": "https://placekitten.com/200/300",
              }
            }
            style={
              Object {
                "height": 0,
                "width": 0,
              }
            } />
        </View>
      </View>
    </View>
    <View
      style={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Array [
              undefined,
            ],
          ]
        }>
        You can zoom into this image
      </Text>
    </View>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        
      </Text>
    </TouchableOpacity>
  </View>
</Modal>
`;

exports[`test renders hidden correctly 1`] = `
<Modal
  animationType="fade"
  onRequestClose={[Function]}
  transparent={true}
  visible={false}>
  <View
    style={undefined}>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "height": 667,
            "width": 375,
          },
        ]
      }>
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "scale": 1,
              },
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }>
        <View
          onLayout={[Function]}
          style={
            Object {
              "height": 0,
              "width": 0,
            }
          }>
          <Image
            source={
              Object {
                "uri": "https://placekitten.com/200/300",
              }
            }
            style={
              Object {
                "height": 0,
                "width": 0,
              }
            } />
        </View>
      </View>
    </View>
    <View
      style={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            undefined,
            Array [
              undefined,
            ],
          ]
        }>
        You can zoom into this image
      </Text>
    </View>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        
      </Text>
    </TouchableOpacity>
  </View>
</Modal>
`;
